DataFormats/HcalRecHit/src/HBHERecHit.cc

```diff
@@ -62,3 +62,22 @@ void HBHERecHit::getMergedIds(std::vector<HcalDetId>* ids) const
         }
     }
 }
+
+
+HcalDetId HBHERecHit::idFront() const {
+  if (isMerged()) {
+    HcalDetId myId(id());
+    if (auxPhase1_ & (1U << HBHERecHitAuxSetter::OFF_COMBINED)) {
+      const unsigned nMerged = CaloRecHitAuxSetter::getField(auxPhase1_,
+							     HBHERecHitAuxSetter::MASK_NSAMPLES,
+							     HBHERecHitAuxSetter::OFF_NSAMPLES);
+      if (nMerged > 0) {
+	const unsigned depth = CaloRecHitAuxSetter::getField(auxHBHE_, 0xf, 0);
+	myId = HcalDetId(myId.subdet(), myId.ieta(), myId.iphi(), depth);
```

Comment:
Here is a faster, sufficient implementation which takes into account the fact that it is not possible to merge 0 rechits:

HcalDetId HBHERecHit::idFront() const
{
    if (auxPhase1_ & (1U << HBHERecHitAuxSetter::OFF_COMBINED))
    {
        const HcalDetId myId(id());
        return HcalDetId(myId.subdet(), myId.ieta(), myId.iphi(), auxHBHE_ & 0xf);
    }
    else
        return id();
}

Making isMerged() inline has its own little drawbacks -- one would have to include HBHERecHitAuxSetter.h into HBHERecHit.h, so that every code using rechits picks up an additional header dependence.